src/components/tools/drawing-manager.js

```diff
@@ -6,42 +6,45 @@
  */
 AFRAME.registerComponent("drawing-manager", {
   init() {
-    this._onComponentInitialized = this._onComponentInitialized.bind(this);
-
     this.el.object3D.visible = false;
     this.drawingToPen = new Map();
   },
 
-  remove() {
-    if (this.drawingEl) {
-      this.drawingEl.removeEventListener("componentinitialized", this._onComponentInitialized);
-    }
-  },
-
-  _onComponentInitialized(e) {
-    if (e.detail.name == "networked-drawing") {
-      this.drawing = this.drawingEl.components["networked-drawing"];
-    }
-  },
-
-  createDrawing() {
-    if (!this.drawingEl) {
-      this.drawingEl = document.createElement("a-entity");
-      this.drawingEl.setAttribute("networked", "template: #interactable-drawing");
-      this.el.sceneEl.appendChild(this.drawingEl);
-
-      this.drawingEl.addEventListener("componentinitialized", this._onComponentInitialized);
-    }
+  destroyDrawing() {
+    this.drawingToPen.delete(this.drawing);
+    this.drawing = null;
+    this.drawingEl = null;
   },
 
   getDrawing(pen) {
     //TODO: future handling of multiple drawings
-    if (this.drawing && (!this.drawingToPen.has(this.drawing) || this.drawingToPen.get(this.drawing) === pen)) {
-      this.drawingToPen.set(this.drawing, pen);
-      return this.drawing;
-    }
-
-    return null;
+    return new Promise((resolve, reject) => {
+      if (!this.drawingEl) {
+        this.drawingEl = document.createElement("a-entity");
+        this.drawingEl.setAttribute("networked", "template: #interactable-drawing");
+        this.el.sceneEl.appendChild(this.drawingEl);
+
+        this.drawingEl.addEventListener(
+          "componentinitialized",
+          e => {
+            if (e.detail.name == "networked-drawing") {
+              this.drawing = this.drawingEl.components["networked-drawing"];
+              this.drawingToPen.set(this.drawing, pen);
+              resolve(this.drawing);
```

Comment:
Seems unfortunate that the async initialization affects things at this level, since this will be called for each start draw / stop draw. No specific suggestion here - but would be interested in thinking thru how we might refrain from `getDrawing` calls until the initialization is done so that it is synchronous.

Reply:
This is only a problem now because drawings can be deleted. Before, this initialization only happened once whenever you spawned a pen for the first time. 
Probably the best way to handle this long term would be if `networked-drawing` wasn't an a-frame component so `componentinitialized` wouldn't need to be waited for, but as it is now, that's the only sane way to actually have networking for something.